PowerPointLabs/PowerPointLabs/NarrationsLab/Storage/NarrationsLabStorageConfig.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using System.Xml.Serialization;
+
+using PowerPointLabs.ActionFramework.Common.Log;
+using PowerPointLabs.NarrationsLab.Data;
+
+namespace PowerPointLabs.NarrationsLab.Storage
+{
+    public static class NarrationsLabStorageConfig
+    {
+        private const string defaultNarrationsStorageFolder = "PowerPointLabs Narrations Access Key Storage";
+        private const string defaultNarrationsStorageFile = "useraccount.xml";
+        private static string defaultApplicationFolderPath = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
+
+        public static string GetAccessKeyStoragePath()
+        {
+            return Path.Combine(defaultApplicationFolderPath, defaultNarrationsStorageFolder);
+        }
+
+        public static string GetAccessKeyFilePath()
+        {
+            return Path.Combine(defaultApplicationFolderPath, defaultNarrationsStorageFolder, defaultNarrationsStorageFile);
+        }
+
+        public static void SaveUserAccount(UserAccount account)
+        {
+            if (!Directory.Exists(GetAccessKeyStoragePath()))
+            {
+                Directory.CreateDirectory(GetAccessKeyStoragePath());
+            }
+          
+            Dictionary<string, string> user = new Dictionary<string, string>()
+            {
+                { "endpoint", account.GetRegion() },
+                {"key",  account.GetKey() }
+            };
+            XElement root = new XElement("user", from kv in user select new XElement(kv.Key, kv.Value));
+            FileStream file = File.Open(GetAccessKeyFilePath(), FileMode.OpenOrCreate);
+            root.Save(file);
+            file.Close();
+        }
+
+        public static void LoadUserAccount()
+        {
+            Dictionary<string, string> user = new Dictionary<string, string>();
+            try
+            {
+                FileStream file = File.Open(GetAccessKeyFilePath(), FileMode.Open);
+                XElement root = XElement.Load(file);
+                foreach (XElement el in root.Elements())
+                {
+                    user.Add(el.Name.LocalName, el.Value);
+                }
+                string key = user.ContainsKey("key") ? user["key"] : null;
+                string endpoint = user.ContainsKey("endpoint") ? user["endpoint"] : null;
+                if (key != null && endpoint != null)
```

Comment:
Is there any error handling supposed to be done if either is null?

Reply:
Edited to delete storage file if either is null. User will then have to log in again upon clicking on settings page.